components/theia/packages/gitpod-extension/src/browser/gitpod-git-token-provider.ts

```diff
@@ -138,7 +138,8 @@ export class GitpodGitTokenProvider {
             const validationResult = await this.tokenValidator.checkWriteAccess(authProvider, repoFullName, tokenResult);
             const hasWriteAccess = validationResult && validationResult.writeAccessToRepo === true;
 
-            if (!hasWriteAccess) {
+            if (hasWriteAccess) {
```

Comment:
FYI: this is just an indicator from the API, it's not the truth, so further checks are required.